src/llamacrawl/api/progress_tracker.py

```
@@ -320,40 +320,39 @@ def _save_state(self) -> None:
+                "files_processed": str(self.metrics.files_processed),
+                "files_failed": str(self.metrics.files_failed),
+                "files_skipped": str(self.metrics.files_skipped),
+                "total_documents": str(self.metrics.total_documents),
+                "documents_ingested": str(self.metrics.documents_ingested),
+                "documents_deduplicated": str(self.metrics.documents_deduplicated),
+                "documents_failed": str(self.metrics.documents_failed),
+                "started_at": self.metrics.started_at.isoformat(),
+                "last_update": self.metrics.last_update.isoformat(),
+            }
+            self._redis.set_progress_state(key, mapping, ttl=7 * 24 * 3600)
         except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)